sympy/printing/preview.py

```diff
@@ -125,16 +125,18 @@ def preview(expr, output='png', viewer=None, euler=True, packages=(),
             }
 
             try:
-                for candidate in candidates[output]:
-                    path = find_executable(candidate)
-                    if path is not None:
-                        viewer = path
-                        break
-                else:
-                    raise SystemError(
-                        "No viewers found for '%s' output format." % output)
+                candidate_viewers = candidates[output]
             except KeyError:
                 raise SystemError("Invalid output format: %s" % output)
+
+            for candidate in candidate_viewers:
+                path = find_executable(candidate)
```

Comment:
Should this be `shutil.which`?

Reply:
Probably, but I didn't want to overlap too much with #19634. Now that that's merged, yes.

Reply:
Updated